### PR TITLE
chore: Documentation

### LINKED ISSUE #305: New Github Pages Deployment Mechanism to Replace Sphinx

### PR DESCRIPTION
**Pull request by malachi-constant**
Closes #305 

- [x] Includes `docs/` and removes unnecessary API docs build 
- [x] Setup new API docs linking
- [x] Getting Started Landing Page rewrite 

**How-tos updated and appended to :**
- [x] #302 
- [x] #303 
- [x] #314 

#### Running Documentation Site Locally
- [Install jekyll](https://jekyllrb.com/docs/installation/)
- `cd docs/`
- `bundle install`
- `bundle exec jekyll serve`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 7678c366669872c07e5cfcf2c595c0b6ecb84fd1
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9111f9e0-86e3-4882-a509-d9709b3c906d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: d303bb71bc8a1c7c1a3ae2e7c873cebc99e89a61
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b0ae5811-90eb-497d-bc04-904d95e2ff73%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 8ddb31f4bce9b412e6ed4cfbdb195afd38cec03f
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b847c590-9eb2-4783-8903-3d7c77325d87%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 30111c74ab9186d654ee3a6b309c747773ef7133
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=491b9bb1-5895-41a4-a10a-3d12d4fa1c52%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 95ccf20133dad1304b8f0804347941e72c8a3c92
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cc194397-b931-49cc-b7a9-97533cea812e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 09b1d3e56a26235d90e675d5108868d6d8723295
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0e61736e-5caf-4a63-9133-05fd0d17f159%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 37d207e7487cf18967a3fdcb92b069f1e778bbbf
* Result: SUCCEEDED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1a936d42-4bac-40c7-9125-9f87ddd91245%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 46c039ddd0c9b1abe8d1f34ae6178b8a6b5637bd
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=42592cb8-61ea-42fd-a3b1-135f7d2aca1c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 799ed9378a36cb20d0c15bba2f8bf147abd4811a
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8c487408-296a-4241-957a-03b0bb06ab76%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: e4dd5b8566dbbad986a0d3f234fc76c2e6a25cf5
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d31c5379-ab9a-426c-81b3-c33429b9bf22%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: aa6a315b705b896e10edbddc32b9bdd3a940e646
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2a50afc2-3969-467d-b722-9206bb3a6320%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: docs/index.markdown [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    scope=self,
+    id="ddk-sqs-lambda",
+    environment_id=environment_id,
+    code=Code.from_asset("./lambda"),
+    handler="index.lambda_handler",
+    layers=[
+        LayerVersion.from_layer_version_arn(
+            self,
+            "ddk-lambda-layer-wrangler",
+            f"arn:aws:lambda:{self.region}:336392948345:layer:AWSDataWrangler-Python39:2",
```
Comment (LeonLuttenberger) [inline docs/index.markdown:67]: What's a `AWSDataWrangler`? Surely you mean `PandasSDK` :)

File: docs/index.markdown [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+DDK comes with a library of stages, however users can also create their own based on their use cases, 
+and are encouraged to share them with the community. 
+
+Let's take a look at an example below:
+
+```python
+...
+
+firehose_s3_stage = KinesisToS3Stage(
```
Comment (LeonLuttenberger) [inline docs/index.markdown:50]: These are the old APIs, right? `KinesisToStage` doesn't have an `environment_id` anymore. Shouldn't we use the new APIs we got from JSII?

File: docs/_site/index.html [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+<a href="https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-rules.html">AWS EventBridge Rules</a> .</p>
+
+<p>DDK comes with a library of stages, however users can also create their own based on their use cases, 
+and are encouraged to share them with the community.</p>
+
+<p>Let’s take a look at an example below:</p>
+
+<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">...</span>
+
+<span class="n">firehose_s3_stage</span> <span class="o">=</span> <span class="n">KinesisToS3Stage</span><span class="p">(</span>
```
Comment (LeonLuttenberger) [inline docs/_site/index.html:137]: This part is autogenerated from `index.markdown` I assume? Do we need to gave these autogenerated HTML files in the repo? Can we at least use `.gitattributes` to [mark them as such](https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github)?

File: docs/release/next/how-to/custom-stage.markdown [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+## The DataStage Class
+
+The AWS DDK provides a construct: [DataStage()](https://constructs.dev/packages/aws-ddk-core/v/1.0.0-beta.1/api/DataStage?lang=typescript) that can be inherited to build your own custom stages.
+
+## Building Your Own Data Stage
+Let's say we need to create a stage that publishes S3 events to an SNS Topic.
+
+We'll first need to create a Stage for the topic.
+
+{% tabs class %}
```
Comment (LeonLuttenberger) [inline docs/release/next/how-to/custom-stage.markdown:18]: Nice! I love these tabs.

File: docs/index.markdown (hunk 2 above)
Comment (malachi-constant) [reply docs/index.markdown:50]: Yea I need to update this thanks.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 116cd5b7d3f1e098252cb87cba9b5419efda895c
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3af1785b-7d32-4699-86b4-81333bdb7fb4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: d6bb035acfb8afdd4a17e714aec1d1da4858776d
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c8fa2545-21a4-4acc-94a7-199146b405ef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 640f04fb45aca5ae75c62d680e14af8716948bd3
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fbfda71f-c58e-43a2-b022-b8d3320591e8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: ca59c0498172243a5dd7a1509aac1823742f8b63
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4f8a66dc-3f86-4190-9878-d6e9eb35c6ce%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: DDKIntegrationTestsD8472609-5SCHWjldmWcf
* Commit ID: 31bd696df3cf95156e8c763bbe1d43778c6dc0e8
* Result: FAILED
* [Build Logs](https://9hm1y4sor9.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a345e5b2-973d-4ae7-9a1d-0ec4e39c7f8e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by LeonLuttenberger — APPROVED** [review]

**Pull request merged**